diff --git a/alsa-ucm-conf/ucm2/codecs/rt5640/init.conf b/alsa-ucm-conf/ucm2/codecs/rt5640/init.conf
new file mode 100644
index 0000000..5c960ce
--- /dev/null
+++ b/alsa-ucm-conf/ucm2/codecs/rt5640/init.conf
@@ -0,0 +1,19 @@
+
+LibraryConfig.remap.Config {
+
+	ctl.default.remap {
+		"name='ADC Boost Gain'" "name='ADC Boost Capture Volume'"
+		"name='IN1 Boost'" "name='IN1 Boost Capture Volume'"
+		"name='IN2 Boost'" "name='IN2 Boost Capture Volume'"
+		"name='IN3 Boost'" "name='IN3 Boost Capture Volume'"
+		# Rename the 'Headphone Switch' DAPM PIN switch to avoid it getting
+		# grouped with 'Headphone Playback Volume'
+		"name='Headphone Switch'" "name='Headphone Output Switch'"
+		"name='HP Playback Volume'" "name='Headphone Playback Volume'"
+		"name='HP Channel Switch'" "name='Headphone Playback Switch'"
+		# Idem for the 'Speaker Switch'
+		"name='Speaker Switch'" "name='Speaker Output Switch'"
+		"name='Speaker Channel Switch'" "name='Speaker Playback Switch'"
+	}
+
+}
diff --git a/alsa-ucm-conf/ucm2/lib/card-init.conf b/alsa-ucm-conf/ucm2/lib/card-init.conf
new file mode 100644
index 0000000..527a5e0
--- /dev/null
+++ b/alsa-ucm-conf/ucm2/lib/card-init.conf
@@ -0,0 +1,10 @@
+#
+# Manage directory for the card specific configuration
+#
+
+Define.LibDir "/var/lib/alsa/card${CardNumber}.conf.d"
+
+FixedBootSequence [
+	exec "/bin/rm -rf ${var:LibDir}"
+	exec "/bin/mkdir -p -m 0755 ${var:LibDir}"
+]
diff --git a/alsa-ucm-conf/ucm2/lib/ctl-remap.conf b/alsa-ucm-conf/ucm2/lib/ctl-remap.conf
new file mode 100644
index 0000000..06e775c
--- /dev/null
+++ b/alsa-ucm-conf/ucm2/lib/ctl-remap.conf
@@ -0,0 +1,22 @@
+#
+# This is generic UCM configuration for alsa-lib
+# Users are not allowed to overwrite this configuration
+#
+
+LibraryConfig.generic.Config {
+
+	ctl.default {
+		@args [ CARD ]
+		@args.CARD.type string
+		type remap
+		child {
+			type hw
+			card $CARD
+		}
+	}
+
+}
+
+FixedBootSequence [
+	cfg-save "${var:LibDir}/ctl-remap.conf:+ctl.default"
+]
